proposals/0684-adding-e2e-in-core.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Introducing E2E testing in React Native core
+author:
+  - Lorenzo Sciandra
+  - Nicola Corti
+  - Riccardo Cipolleschi
+  - <Callstack folks>
+date: 2023-07-06
+---
+
+# RFC0684: Introducing E2E testing in React Native core
+
+## Summary
+
+Testing is how we validate that the code is good, and ensure that what we build our products on top of is solid and our customers are happy.
+
+For the longest time, in React Native core this validation has been done in a few ways:
+
+- via Meta using the source code in their own products first
+- via internal testing with custom tooling in the Meta monorepo
+- via (mostly) unit testing on the CircleCI GitHub `react-native` repo
+- via local testing by the release crew ahead of generating new releases
+- via "pre-stable" releases (AKA Release Candidates and nightlies) that members from the community could use to test and report back
+
+There is a great opportunity for improvement here, by introducing End-to-End (E2E) testing in the CircleCI GitHub `react-native` repository to introduce one more level of validation that aims to replace the need for (at least) the local testing from the release crew. By adding tooling that spins up the RNTester app in the codebase, and "black box" (the RNTester app is not aware that the agent using it is not a human) testing it, we can trust that the code is more solid.
+
+This will make the codebase more stable, surface issues rapidly to the commit authors, and reduce the workload for the release crew.
+
+## Basic example
+
+On a basic level, the idea is to have fully automated end-to-end testing running on CircleCI on `react-native`'s every single commit to flag if any code change is inadvertently introducing a regression of some form.
+
+The code and tests to do so will live in a new dedicated monorepo package, `packages/rn-tester-e2e` and rely on the existing `RNTester` test app.
+
+The tooling combination selected for this is composed of [Appium](https://appium.io/) and [WebDriverIO](https://webdriver.io/) and [Jest](https://jestjs.io/).
+
+## Motivation
+
+The main "why" has already been explained above: introducing more solid automation to thoroughly verify that code changes don't introduce regressions. This will lead to removing the need for the release crew to do local testing to validate the code before a release.
+
```

Comment:
As an additional reason, this can save release time, as we don't have to test the release manually. This can produce more stabler releases of react native.